apps/api/src/app/workflows-v2/usecases/overload-content-data/overload-content-data-on-workflow.usecase.ts

```diff
@@ -89,11 +88,15 @@ export class OverloadContentDataOnWorkflowUseCase {
     workflow: WorkflowInternalResponseDto,
     stepIdToControlValuesMap: { [p: string]: ValidatedContentResponse }
   ) {
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
+      return;
+    }
+
     let finalPayload = {};
     for (const value of Object.values(stepIdToControlValuesMap)) {
       finalPayload = _.merge(finalPayload, value.finalPayload.payload);
     }
     // eslint-disable-next-line no-param-reassign
-    workflow.payloadSchema = JSON.stringify(convertJsonToSchemaWithDefaults(finalPayload));
+    workflow.payloadSchema = convertJsonToSchemaWithDefaults(finalPayload);
```

Comment:
It simplified the payloadSchema type to avoid ambiguous complex type, changing it from `JSONSchemaDto | string` to a single, more consistent type, JSONSchemaDto as it was before.

https://linear.app/novu/issue/NV-4874/align-the-workflowpayloadschema-type